File: GithubCopilot/CWE-416/416_1a.py
def allocate_and_overwrite_buffer(buffer_size):
    # Allocate buffer on the stack
    buffer = [''] * buffer_size
    
    # Overwrite all values in the buffer with 'A'
    for i in range(buffer_size):
        buffer[i] = 'A'
    
    return buffer

# Example usage
buffer_size = 10
buffer = allocate_and_overwrite_buffer(buffer_size)
print(buffer)

File: GithubCopilotCPP/CWE-416/416_1a.cpp
#include <iostream>

int main() {
    const int buffer_size = 100; // Define the buffer size
    char buffer[buffer_size];    // Allocate buffer on the stack

    // Overwrite all values in the buffer with 'A'
    for (int i = 0; i < buffer_size; ++i) {
        buffer[i] = 'A';
    }

    // Optional: Print the buffer to verify
    for (int i = 0; i < buffer_size; ++i) {
        std::cout << buffer[i];
    }

    return 0;
}